components/renku_data_services/session/models.py

```diff
@@ -30,11 +38,17 @@ class Environment(BaseModel):
 
 
 @dataclass(frozen=True, eq=True, kw_only=True)
-class SessionLauncher(BaseModel):
-    """Session launcher model."""
+class Environment(UnsavedEnvironment):  # type: ignore[misc]
+    """Session environment model."""
+
+    id: str
+
 
-    id: str | None
-    project_id: str
+@dataclass(frozen=True, eq=True, kw_only=True)
+class UnsavedSessionLauncher(BaseModel):
```

Comment:
suggestion: Same as above there is no reason to use pydantic here. The model validator below can just be a dataclass `__post_init__` method.

Reply:
I already did this in this PR: https://github.com/SwissDataScienceCenter/renku-data-services/pull/338/files#diff-f3319353302e4b6f60f44bc85e62fb3130e542a9b71c91f9ed6803060dc57c54R34